config.yaml.example

```diff
@@ -19,7 +19,7 @@ install:
   #     size: 300
 
   # default partitions
-  # only 'oem', 'recovery', 'state' and 'persistent' objects allowed
+  # only 'efi', 'oem', 'recovery', 'state' and 'persistent' objects allowed
```

Comment:
Just nitpicking, but it would be best to also include an efi partition example, just to highlight the fact that the filesystem to use should be `vfat`. Or can it be omitted even? I don't truly know if we use the defaults or not.

Reply:
vfat filesystem is set by default and I was wondering to force it or not (error out if not set to vfat), but then I thought about better giving some flexibility as this is pretty hidden in the setup and if one is actually touching this is probably for a good reason. For instance I don't know if for RPi it is mandatory to boot from vfat or it can be another filesystem.
What we probably could consider in the future is renaming this EFI partition to some firmware neutral name (boot partition?), so there is no explicit assumption we are using EFI firmware. But I did not want to touch that at this time because it feels like a risky change, there might be some corner cases hard to foresee.

Reply:
I would not force it as well, but it could be an idea. What I missed is an example for the end-users to follow, let's say I do want to use this feature, then I would miss (without looking at the source code) how can I extend the partition size, and what do I need to configure in order to do that.
I was wondering is that `fat`, `fat32`, etc. before looking at the source, and if it's optional is even better, so I could just code:

```yaml
efi:
  size: 1000
```

Reply:
Indeed this is the case you can provide only size and this will only modify the default size, but keep all other parameters. About partition sizes we definitely need some docs on that topic and we already have an issue for that https://github.com/rancher/elemental-docs/issues/345